### PR TITLE
ICU-21581 lingering exhaustive fail, need to separately skip sd_Deva month name exemplar test

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21581
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

Lingering exhaustive test failure, need to separately skip the month names exemplar test (for sd_Deva) in addition to the day names exemplar test (for mai, sd_Deva)

### CODE REVIEW

**Review by sffc — APPROVED** [review]

**Comment by sffc** [comment]
C++ warning is blocking CI:

```
cldrtest.c:1034:54: error: size argument in 'strncmp' call is a comparison [-Werror,-Wmemsize-comparison]
                if (uprv_strncmp(currLoc,"sd_Deva",7 == 0) && 
                                                   ~~^~~~
../../common/cstring.h:44:74: note: expanded from macro 'uprv_strncmp'
#define uprv_strncmp(s1, s2, n) U_STANDARD_CPP_NAMESPACE strncmp(s1, s2, n)
                                                                         ^
cldrtest.c:1034:21: note: did you mean to compare the result of 'strncmp' instead?
                if (uprv_strncmp(currLoc,"sd_Deva",7 == 0) && 
                    ^
../../common/cstring.h:44:58: note: expanded from macro 'uprv_strncmp'
#define uprv_strncmp(s1, s2, n) U_STANDARD_CPP_NAMESPACE strncmp(s1, s2, n)
                                                         ^
cldrtest.c:1034:52: note: explicitly cast the argument to size_t to silence this warning
                if (uprv_strncmp(currLoc,"sd_Deva",7 == 0) && 
                                                   ^
                                                   (size_t)(
../../common/cstring.h:44:74: note: expanded from macro 'uprv_strncmp'
#define uprv_strncmp(s1, s2, n) U_STANDARD_CPP_NAMESPACE strncmp(s1, s2, n)
                                                                         ^
```

File: icu4c/source/test/cintltst/cldrtest.c [hunk 1]
```diff
@@ -1031,6 +1031,10 @@ static void VerifyTranslation(void) {
                 else {
                     end = ures_getSize(resArray);
                 }
+                if (uprv_strncmp(currLoc,"sd_Deva",7 == 0) &&
```
Comment (sffc) [inline icu4c/source/test/cintltst/cldrtest.c:1034]:
```suggestion
                if (uprv_strncmp(currLoc,"sd_Deva",7) == 0 && 
```

**Review by sffc — CHANGES_REQUESTED** [review]

**Comment by pedberg-icu** [comment]
Shane, thanks for the paren fix! My local make check build seemed to succeed even in spite of that... anyway, rolled that in and squashed.

**Review by jefgen — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Comment by srl295** [comment]
CLDR-14995

**Pull request merged**